internal/packages/internal/packagestructure/conversion.go

```diff
@@ -0,0 +1,92 @@
+package packagestructure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+
+	"package-operator.run/internal/apis/manifests"
+	"package-operator.run/internal/packages/internal/packagetypes"
+)
+
+type manifestConstraint interface {
+	manifests.PackageManifest | manifests.PackageManifestLock
+}
+
+func ManifestFromFile[T manifestConstraint, PT interface {
+	runtime.Object
+	*T
+}](
+	_ context.Context, scheme *runtime.Scheme,
+	path string, manifestBytes []byte,
+) (*T, error) {
+	manifest := PT(new(T))
```

Comment:
```suggestion
	manifest := PT(&T{})
```
that would be the same, right? Not requiring this suggestion, honestly just want to know :D